### PR TITLE
Miming Refactor

### PR DESCRIPTION
**Pull request by Fox-McCloud**
Refactors miming a bit; largely from TG

Fixes
- Miming actually prevents you from speaking
- Can't use your wall if you're not miming

Changes
- Mimes now have two spells instead of verbs: the invisible wall and speaking
 - These have unique action button icons and the likes; also lets you know when they're off cooldown
- Adds in support for a few new invocation types in spells (mime spells use these)
- Forefields now prevent atmos from passing over them
- Mime walls now last a duration of 300 instead of 50

Coder Stuffs
- Uses a new `IsVocal` proc; this is used for miming, for now, but could be used to check if certain conditions are met to prevent/allow someone to speak (but still emote or use hiveminds).

:cl: Fox McCloud
bugfix: Mimes can no longer use spells and continue speaking
tweak: Mime abilities are now spells with proper icons
tweak: Mime wall now last 30 seconds, up from 5
tweak: forecfields/invisible walls now block air currents
rscadd: Adds the Sleeping Carp scroll to the uplink for 17 TC
bugfix: Fixes Shadowlings being able to use guns
bugfix: Fixes sleeping carp scroll users being able to use guns
/:cl:

*cough* um...including the sleeping crap scroll PR changelog here because...I forgot...*cough*

### CODE REVIEW

**Comment by Fox-McCloud** [comment]
Oh what the hell

**Comment by Spacemanspark** [comment]
Everything broke.

**Comment by Fox-McCloud** [comment]
>Everything broke.

No, a file got unchecked somehow. NO idea how.

File: code/modules/mob/living/carbon/human/say.dm [hunk 1]
```diff
@@ -106,7 +111,7 @@
 	var/list/returns[3]
 	var/speech_problem_flag = 0
 
-	if(silent || (sdisabilities & MUTE))
+	if(silent || (sdisabilities & MUTE) || !IsVocal())
```
Comment (tigercat2000) [inline code/modules/mob/living/carbon/human/say.dm]: Yeah, no.
This goes in it's own conditional, in https://github.com/ParadiseSS13/Paradise/blob/master/code/modules/mob/living/say.dm#70

File: code/modules/mob/living/carbon/human/emote.dm [hunk 2]
```diff
@@ -21,6 +21,10 @@
 		if (I.implanted)
 			I.trigger(act, src)
 
+	var/miming=0
```
Comment (tigercat2000) [inline code/modules/mob/living/carbon/human/emote.dm]: spacing

**Comment by TheDZD** [comment]
Changeloooooogs

**Comment by Aurorablade** [comment]
is it supposed to actually make a wall..cuase i don't see where its making a wall..no i am not making a pun i am serious.

**Comment by Fox-McCloud** [comment]
@Aurorablade 

Right here: https://github.com/ParadiseSS13/Paradise/pull/3736/files#diff-43170e18cc250943661efb80105dc003R6

It's the way summon spells work. You can make summon spells literally summon anything.

**Comment by Regen1** [comment]
EHHHH

30 second wall is WAY to strong, this makes escaping in areas like maintenance as easy as pressing a button

5 seconds is fine for whats supposed to be a MINOR spell / joke

**Comment by Robertha89** [comment]
Whops, trapped myself for 30 seconds, considering the mime wall has been bugged for a longer time during the life of Paradise then it has been working, this seems to be a nice thing to have, if it is shows to be as stupidly strong as we think, then it can be reduced.

**Comment by TheDZD** [comment]
If memory serves, I fixed the mime wall spell a long while ago so it actually worked, but decreased the timer from 30 to 5 seconds at the recommendation of a maintainer (can't remember who it was) due to concerns it would be too long-lasting.

**Comment by Robertha89** [comment]
it always lasted for 3 seconds, even after you fixed it, it worked for a while been broken for a year at least.

File: code/modules/mob/living/say.dm [hunk 3]
```diff
@@ -88,9 +90,12 @@ proc/get_radio_key_from_channel(var/channel)
 	if(GREY in mutations)
 		message = "<span class='grey'>[message]</span>"
 
-	else if(COMIC in mutations)
+	if(COMIC in mutations)
 		message = "<span class='sans'>[message]</span>"
 
+	if(!IsVocal())
+		message = ""
```
Comment (VampyrBytes) [inline code/modules/mob/living/say.dm]: @Fox-McCloud Won't this stop them emoting as well?

Comment (VampyrBytes) [inline code/modules/mob/living/say.dm]: Nevermind, found the answer (no)

File: code/modules/mob/living/say.dm [hunk 4]
```diff
@@ -88,9 +90,12 @@ proc/get_radio_key_from_channel(var/channel)
 	if(GREY in mutations)
 		message = "<span class='grey'>[message]</span>"
 
-	else if(COMIC in mutations)
+	if(COMIC in mutations)
```
Comment (tigercat2000) [inline code/modules/mob/living/say.dm]: why did you remove this else condition
I understand you had trouble with getting IsVocal to work, however, what does this have to do with it? An else here should only be part of the if statement above, "if grey in mutations, else if comic in mutations" followed by a regardless "if not vocal"

Comment (Fox-McCloud) [inline code/modules/mob/living/say.dm]: Functionally, I don't see the point to it--regardless of `if` or `else if`, the comic mutation is going to override the grey mutation if someone ultimately has both.

Comment (tigercat2000) [inline code/modules/mob/living/say.dm]: @Fox-McCloud the point is making sure that only one class is applied, to avoid inheritance issues

sure right now maybe comic overrides what the grey mutation does, but what about when it doesn't?

Comment (Tastyfish) [inline code/modules/mob/living/say.dm]: Is there any sanitization or something that would break if it stacked multiple spans? Speaking in a foreign language on a radio channel with grey or comic already stacks 4 spans iirc.

Comment (TheDZD) [inline code/modules/mob/living/say.dm]: Evidently, some things such as stammering from pain do break the comic span.

**Comment by Regen1** [comment]
@Robertha89 So because it's been broken for so long we're going to buff it to ridiculous lengths? That's not how balancing works and sounds more like a "I want a role I like to be stronger" argument

**Comment by TheDZD** [comment]
@Robertha89 Except, it isn't broken. It creates the wall just fine, I managed to test this on a local server, and was able to shoot at the invisible wall and have it block laser shots as it is supposed to.

**Comment by Fox-McCloud** [comment]
Alright, not commenting on "yes, keep it should be this, no it should be that"--just some objective facts here.

- The Mime has had a wall spell for eons
- The Mime's wall spell was originally 30 seconds and had been 30 seconds ever since he had the spell
- At one point in time, the spell broke and the placed wall didn't actually do anything (it did at one point; dunno what broke it)
- The wall was eventually fixed; when it was fixed, the time was lowered from 30 seconds to 5 seconds
- Enter this PR.

File: code/modules/mob/living/say.dm (hunk 4 above)
Comment (Fox-McCloud) [inline code/modules/mob/living/say.dm]: There's some condition that triggers this---as I was touching this code, I tried to replicate this by giving myself comic, slurring, stuttering, hulk, and a number of others. I then talked locally and over radio, but I couldn't trigger the telltale "<s-span-class="wa-r-rn-ning> I k-kne-w I s-shouildn't d-do that-t-t </s-spa-nn>"
